src/openms/source/FILTERING/DATAREDUCTION/IsotopeDistributionCache.cpp

```diff
@@ -40,89 +40,116 @@
 namespace OpenMS
 {
 
-  IsotopeDistributionCache::IsotopeDistributionCache(double max_mass, double mass_window_width, double intensity_percentage, double intensity_percentage_optional) :
-    mass_window_width_(mass_window_width)
+  IsotopeDistributionCache::IsotopeDistributionCache()
   {
-    Size num_isotopes = std::ceil(max_mass / mass_window_width) + 1;
+    mass_window_width_ = 20.0;
+    intensity_percentage_=0.0;
+    intensity_percentage_optional_ =0.0;
 
-    //reserve enough space
-    isotope_distributions_.resize(num_isotopes);
+  }
+
+
+  void IsotopeDistributionCache::calculateisotopeDistribution(OpenMS::Size num_begin)
```

Comment:
`calculateisotopeDistribution` is a bit unspecific.
You are effectively extending the cache?